### PR TITLE
Use SPMI_CACHE_DIRECTORY for sharing mch files across repos

### PR DESCRIPTION
**Pull request by kunalspathak**
If `SPMI_CACHE_DIRECTORY ` is set to a directory, `superpmi.py` will use that location to download mch files. That way multiple repositories can share the .mch files instead of downloading them for every repository.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -3022,7 +3023,12 @@ def setup_args(args):
                         "Unable to set log_file.")
 
     def setup_spmi_location_arg(spmi_location):
-        return os.path.abspath(os.path.join(coreclr_args.artifacts_location, "spmi")) if spmi_location is None else spmi_location
+        if spmi_location is None:
+            if "SPMI_LOCATION" in os.environ:
+                spmi_location = os.environ["SPMI_LOCATION"]
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:3028]: Can you add:
```
spmi_location = os.path.abspath(spmi_location)
```
to force an absolute path?

File: src/coreclr/scripts/superpmi.py [hunk 2]
```diff
@@ -147,6 +147,7 @@
 spmi_location_help = """\
 Directory in which to put SuperPMI files, such as downloaded MCH files, asm diffs, and repro .MC files.
 Optional. Default is 'spmi' within the repo 'artifacts' directory.
+If 'SPMI_LOCATION' environment variable is set to a path, it will use that directory.
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:150]: I think this is fine, and matches the argument. But I wonder if we should be more verbose and consistent for environment variables, and use `SUPERPMI_LOCATION`.

I never liked the "location" name, but I don't have a better idea. `SUPERPMI_CACHE_DIRECTORY`?

**Review by BruceForstall — CHANGES_REQUESTED** [review]

File: src/coreclr/scripts/superpmi.py (hunk 2 above)
Comment (kunalspathak) [reply src/coreclr/scripts/superpmi.py:150]: sure, may be `SUPERPMI_CACHE_DIR` to keep it short?

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi.py:150]: actually, i would just keep it `SUPERPMI_CACHE_DIRECTORY`. Sounds better and clear.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**